### PR TITLE
E2E kubectl util refactor

### LINKED ISSUE #337: Refactor kubectl e2e util
The way our kubectl util is now does not allow to work with multiple kubeconfig to work on multicluster test related. We need to improve the util and allow the e2e framework in some way to set which is the cluster where the action is going to be performed

### PR DESCRIPTION
**Pull request by fjglira**
#### What type of PR is this?

- [ ] Enhancement / New Feature
- [ ] Bug Fix
- [ ] Refactor
- [ ] Optimization
- [X] Test
- [ ] Documentation Update

#### What this PR does / why we need it:
I'm refactoring the kubectl util to be able to handle better multiple kubeconfig. 

#### Which issue(s) this PR fixes:

Fixes #337

### CODE REVIEW

File: tests/e2e/util/kubectl/kubectl.go [hunk 1]
```diff
@@ -21,97 +21,91 @@ import (
...
 }
 
-// ApplyString applies the given yaml string to the cluster
-func ApplyString(ns, yamlString string, kubeconfig ...string) error {
-	nsflag := nsflag(ns)
-	// If the namespace is empty, we need to remove the flag because it will fail
-	// TODO: improve the nsflag function to handle this case
-	if ns == "" {
-		nsflag = ""
+func (k *KubectlBuilder) setKubeconfig() {
```
Comment (dgn) [inline tests/e2e/util/kubectl/kubectl.go:52]: do we really need this func? kubeconfig fields will be empty after initialization, right?

File: tests/e2e/util/kubectl/kubectl.go [hunk 2]
```diff
@@ -21,97 +21,91 @@ import (
...
 
-// Apply applies the given yaml file to the cluster
-func Apply(ns, yamlFile string, kubeconfig ...string) error {
-	err := ApplyWithLabels(ns, yamlFile, "", kubeconfig...)
-	return err
+// NewKubectlBuilder creates a new KubectlBuilder
+func NewKubectlBuilder() *KubectlBuilder {
+	k := newKubectlBuilder()
+	k.setBinary()
+	k.setKubeconfig()
```
Comment (dgn) [inline tests/e2e/util/kubectl/kubectl.go:80]: this doesn't do anything, does it?

File: tests/e2e/util/kubectl/kubectl.go [hunk 3]
```diff
@@ -21,97 +21,91 @@ import (
...
 
-// CreateFromString creates a resource from the given yaml string
-func CreateFromString(yamlString string, kubeconfig ...string) error {
-	cmd := kubectl("create %s -f -", optionalKubeconfig(kubeconfig))
-	_, err := shell.ExecuteCommandWithInput(cmd, yamlString)
-	if err != nil {
-		return fmt.Errorf("error creating resource from yaml: %w", err)
-	}
-	return nil
+func (k *KubectlBuilder) setNamespace() {
```
Comment (dgn) [inline tests/e2e/util/kubectl/kubectl.go:48]: same here, I don't think we need this func

**Review by dgn — COMMENTED** [review]
Refactor looks good to me, thank you @fjglira! Just a few minor things

File: tests/e2e/util/kubectl/kubectl.go (hunk 2 above)
Comment (fjglira) [reply tests/e2e/util/kubectl/kubectl.go]: Updated by deleting `setNamespace()` and `setKubeconfig()`, now the NewKubectlBuilder func will return the `KubectlBuilder` with only set the Binary from the env var

**Review by dgn — APPROVED** [review]

**Pull request merged**